src/server/kbn_server.js

```diff
@@ -142,17 +139,23 @@ export default class KbnServer {
   async listen() {
     await this.ready();
 
+    const { server, config } = this;
+
+    await new KibanaMigrator({ kbnServer: this }).migrateIndex();
+
+    await fromNode(cb => server.start(cb));
```

Comment:
__note:__ I removed `start` call in https://github.com/elastic/kibana/pull/22190 as we didn't find any reason to keep it in the legacy world, let me know if you still need it for some reason.

Reply:
Hm. I only need a place to put migrations *prior* to Kibana serving HTTP requests. I don't really care where / what it is called. Is there a new place where I should place that?

Reply:
I think you can keep where it's, legacy Kibana won't accept any requests until `kbnServer.ready()` is completed (core will be accepting requests though, but I don't think it can a be problem for you right now). 

In the comment above I just meant that you mistakenly returned `server.start` during rebase/master-merge or something, because [we don't have it in master](https://github.com/elastic/kibana/blob/master/src/server/kbn_server.js#L142-L157) anymore. That presumably caused the error @epixa was observing.

Reply:
I just tested this out after the last commit and it looks to have resolved the issue.